x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/HealthApiFeatureSetUsage.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * Models the health api usage section in the XPack usage response. A sample response would look like this:
+ * {
+ *   "enabled": true,
+ *   "available": true,
+ *   "invocations": {
+ *     "total": 22,
+ *     "verbose_true": 12,
+ *     "verbose_false": 10
+ *   },
+ *   "statuses": {
+ *     "green": 10,
+ *     "yellow": 4,
+ *     "red": 8,
+ *     "values": ["green", "yellow", "red"]
+ *   },
+ *   "indicators": {
+ *     "red" : {
+ *       "master_stability": 2,
+ *       "ilm":2,
+ *       "slm": 4,
+ *       "values": ["master_stability", "ilm", "slm"]
+ *     },
+ *     "yellow": {
+ *       "disk": 1,
+ *       "shards_availability": 1,
+ *       "master_stability": 2,
+ *       "values": ["disk", "shards_availability", "master_stability"]
+ *     }
+ *   },
+ *   "diagnoses": {
+ *     "red": {
+ *       "elasticsearch:health:shards_availability:primary_unassigned": 1,
+ *       "elasticsearch:health:disk:add_disk_capacity_master_nodes": 3,
+ *       "values": [
+ *         "elasticsearch:health:shards_availability:primary_unassigned",
+ *         "elasticsearch:health:disk:add_disk_capacity_master_nodes"
+ *       ]
+ *     },
+ *     "yellow": {
+ *       "elasticsearch:health:disk:add_disk_capacity_data_nodes": 1,
+ *       "values": [""elasticsearch:health:disk:add_disk_capacity_data_nodes"]
+ *     }
+ *   }
+ * }
+ */
+public class HealthApiFeatureSetUsage extends XPackFeatureSet.Usage {
+
+    private final Map<String, Object> usageStats;
+
+    public HealthApiFeatureSetUsage(StreamInput in) throws IOException {
+        super(in);
+        usageStats = in.readMap();
+    }
+
+    public HealthApiFeatureSetUsage(
+        boolean available,
+        boolean enabled,
+        Counters stats,
+        Set<HealthStatus> statuses,
+        Map<HealthStatus, Set<String>> indicators,
+        Map<HealthStatus, Set<String>> diagnoses
+    ) {
+        super(XPackField.HEALTH_API, available, enabled);
+        usageStats = stats.toNestedMap();
+        addValues(usageStats, List.of("statuses"), statuses.stream().map(HealthStatus::xContentValue).collect(Collectors.toSet()));
+        for (HealthStatus status : indicators.keySet()) {
+            addValues(usageStats, List.of("indicators", status.xContentValue()), indicators.get(status));
+        }
+        for (HealthStatus status : diagnoses.keySet()) {
+            addValues(usageStats, List.of("diagnoses", status.xContentValue()), diagnoses.get(status));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void addValues(Map<String, Object> map, List<String> path, Set<String> values) {
+        if (values.isEmpty()) {
+            return;
+        }
+        Map<String, Object> currentMap = map;
+        for (String field : path) {
+            currentMap = (Map<String, Object>) currentMap.computeIfAbsent(field, k -> new HashMap<>());
+        }
+        currentMap.put("values", values.stream().sorted().toList());
+    }
```

Comment:
Shall we just do a `currentMap.put` wherever this is called? It seems to be a shallow method

Reply:
It's not clear to me what you mean. Can you give me an example?

Reply:
Ah sorry, so `addValues` seems to be a bit shallow and I was wondering if we actually need it or could we just replace it with `map.computeIfAbsent( ... ).put("values", valuesList)` ? 

It's super minor so feel free to ignore

Reply:
Actually, I see we do indeed pass multiple paths to it. So it makes sense to have it as otherwise we'll need to loop through paths everywhere! 
🚀